### PR TITLE
Feat/tracks on react

### PR DESCRIPTION
**Pull request by halconel**

### CODE REVIEW

File: app/javascript/components/Places/Form/styles.module.scss [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+  align-items: center;
+
+  > .input {
+    max-width: 10rem;
+  }
+
+  > :not(:last-child) {
+    margin-right: 0.5rem;
+  }
+}
```
Comment (skyksandr) [inline app/javascript/components/Places/Form/styles.module.scss:30]: Looks like it is unused.

File: app/javascript/components/Places/Form/validationSchema.js [hunk 2]
```diff
@@ -0,0 +1,12 @@
+import * as Yup from 'yup'
+
+const schema = Yup.object().shape({
+  name: Yup.string().required('This field is required'),
```
Comment (skyksandr) [inline app/javascript/components/Places/Form/validationSchema.js:4]: I would suggest to use `I18n` here to show correct translations.
Like:
```
import { I18n } from 'components/TranslationsProvider'
...
name: Yup.string().required(() => I18n.t('activerecord.errors.places...')
```
Can't pull translation key straight out of my head, but I'm sure it is easy to find

File: app/javascript/components/Places/NewPlace/index.js [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+import { useI18n } from 'components/TranslationsProvider'
+
+const NewPlace = () => {
+  const newEventMutation = useNewPlaceMutation()
+  const history = useHistory()
+
+  const createEvent = async values => {
+    try {
+      const { data: place } = await newEventMutation.mutateAsync(values)
+      console.log(place)
```
Comment (skyksandr) [inline app/javascript/components/Places/NewPlace/index.js:16]: Looks like a leftover

File: app/javascript/components/Places/NewPlace/index.js [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+      history.push(`/places/${place.id}`)
+    } catch (err) {
+      console.warn(err)
```
Comment (skyksandr) [inline app/javascript/components/Places/NewPlace/index.js:19]: I know you were working by example, but maybe we would better show error before the form?

File: app/javascript/components/Places/index.js [hunk 5]
```diff
@@ -1,16 +1,24 @@
...
+
   return (
     <AppShell fullScreen>
       <Switch>
         <Route exact path="/places" component={PlacesIndex} />
-        <Route path="/places/:id" component={Place} />
+        <Route exact path="/places/new">
+          {canCreatePlace ? <NewPlace /> : <Redirect to="/places" />}
+        </Route>
+        <Route path="/places/:id(\d+)" component={Place} />
```
Comment (skyksandr) [inline app/javascript/components/Places/index.js:21]: My suggestion would be:
```
 {canCreatePlace && <Route exact path="/places/new" component={NewPlace} >}
<Route path="/places/:id(\d+)" component={Place} />
<Route component={() => <Redirect to="/places" />} />
```

**Review by skyksandr — APPROVED** [review]
See comments. 
Other than that - Well Done 👍

File: app/javascript/components/Places/NewPlace/index.js (hunk 3 above)
Comment (halconel) [reply app/javascript/components/Places/NewPlace/index.js:16]: Looks like my brand new tests affected by this leftover too

File: app/javascript/api/hooks/places/place.js [hunk 6]
```diff
@@ -58,3 +59,18 @@ export const usePlaceQueries = ids => {
   const queryClient = useQueryClient()
   return useQueries(ids.filter(Boolean).map(id => placeQuery(id, queryClient)))
 }
+
+const queryKey = id => ['place', id]
+
+const createEvent = place =>
```
Comment (skyksandr) [inline app/javascript/api/hooks/places/place.js:65]: 1. `createPlace`
2. I usually group methods that talk to backend together.

File: app/javascript/components/Places/index.js [hunk 7]
```diff
@@ -1,16 +1,23 @@
...
+import { Route, Switch, Redirect } from 'react-router-dom'
 
 import AppShell from 'components/AppShell'
 import PlacesIndex from 'components/Places/PlacesIndex'
 import Place from 'components/Places/Place'
+import NewPlace from './NewPlace'
+import { useCurrentUserQuery } from 'api/hooks/sessions'
 
 const Places = () => {
+  const { data: currentUser } = useCurrentUserQuery()
```
Comment (skyksandr) [inline app/javascript/components/Places/index.js:11]: Add isLoading here like: `const { data: currentUser, isLoading }`

File: app/javascript/components/Places/index.js [hunk 8]
```diff
@@ -1,16 +1,23 @@
...
+  const canCreatePlace = currentUser?.permissions.canCreatePlace
+
   return (
     <AppShell fullScreen>
```
Comment (skyksandr) [inline app/javascript/components/Places/index.js:15]: Add something like `{isLoading ? (<Loading />) : (...)}`

**Review by skyksandr — APPROVED** [review]

File: config/locales/models/places/en.yml [hunk 9]
```diff
@@ -8,8 +8,15 @@ en:
       place:
         name: Name
         country: Country
+        placeholder: 'Select country'
```
Comment (skyksandr) [inline config/locales/models/places/en.yml:11]: As it is not attribute of a model I'd suggest to move it under `views`

File: app/javascript/components/Places/NewPlace/index.js (hunk 4 above)
Comment (halconel) [reply app/javascript/components/Places/NewPlace/index.js:19]: maybe not? `err` is technical and not for user's eyes

**Review by skyksandr — APPROVED** [review]

**Pull request merged**